contrib/unity/README

```diff
@@ -0,0 +1,285 @@
+Overview
+----------
+TL; DR: Be nice *and* keep Bitcoin in one piece!
+
+Chain splitting can be prevented using the BitcoinUnity soft
+fork and thus guarantee a single Bitcoin future.
+............................................................
+
+Hello dear Bitcoin community,
+
+November 17 is expected to be the activation date of the
+SegWit2X hard fork. This, together with the SegWit part, is
+one of the the most controversial changes to the Bitcoin
+protocol so far.
+
+The next division and fight for the name Bitcoin have the
+possibility of inflicting massive damage to Bitcoin.
+
+I, as a developer in a small bitcoin mine, have done my best
+now to find a remedy for this potential situation.
+
+As you know, there have been several attempts to implement
+replay protection in Bitcoin. They all failed due to
+different obstacles and problems. They failed especially in
+the rightful demand by btc1, that has most of the miners
+behind it, to be Bitcoin and not a split-off.
+
+From the circles of the btc1 project, I got hold of the
+ideas "b0rg" as well as the proposal "BitcoinUnity".
+
+The "b0rg" proposal seems too aggressive.
+
+But the BitcoinUnity project appears to be much more
+practicable.
+
+In this set of patches, you find a complete implementation
+of the BitcoinUnity soft fork idea. The code is currently
+less than 400 lines long and can therefore hopefully be
+fully reviewed in time still.
+
+BitcoinUnity
+------------
+The core idea of BitcoinUnity is to put only those
+transactions into the 1x chain which also succeeded on the
+2x chain, and therefore map a subset of the transactions in
+the 2x chain into the 1x chain.  Preferably those who pay
+maximum transaction fees.
+
+(This does not apply for the separate coinbase transaction
+in 1x)
+
+Ths ensures that a division of the two chains and therefore
+the price and a dirty fight over supremacy of legacy against
+2x Bitcoin is impossible!
+
+At the same time, however, short-term compatibility with
+*all* existing software using the current legacy is
+guaranteed!  (In the long term, this depends on the interest
+of the Miners to keep 1x running)
+
+And in the case that the 2x hard fork is finally not
+desired, this soft fork can be turned off directly and then
+allows habitual, continuous operation of the SegWit1x chain
+like before.
+
+For users and supporters of the 1x chain nothing changes in
+that case, except some, non-essential (and in any case
+expected) hiccups in confirmation time.
+
+Also, the only short-term disadvantage for users of the 1x
+chain is the delay of transaction confirmation by a few
+confirmations on the 2x chain.
+
+Due to the low levels of hashrates initially expected on the
+1x chain and in these times of uncertainty, this should not
+represent a major problem.
+
+The fork is implemented in two variants: UNITY and
+UNITY0. The only difference from UNITY0 to UNITY is that for
+UNITY0, it is expected that all coinbase outputs are zero on
```

Comment:
Would this prevent any tx being created that could only appear on the 1x side? Counter intuitively, aren't even zero value outputs spendable? Maybe they should be zero value *and* sent to `OP_FALSE` (burned).

Reply:
> You've implemented two-way replay double-spend protection via a cross-chain consensus rule soft-fork?

Yessir!

> Would this prevent any tx being created that could only appear on the 1x side? Counter intuitively, aren't even zero value outputs spendable? 

Yes, the zero outputs are valid according to *current rules*. However, they can not be spent according to the rules of the soft fork because any TXIDs involving them would not be whitelisted (or only if such a funny zero-output would exist on the 2x chain). So then that should be fine, I suppose?

Reply:
I thought it would remove even the possibility of creating a tx that could only exist on the 1x chain. As you say, the cross-chain consensus rule would prevent any child of these coinbase outputs from being mined anyway.

It would maybe make more sense for these zero value outputs to point to an address that the miner owns. That way they could potentially be compensated by miners on the 2x side.